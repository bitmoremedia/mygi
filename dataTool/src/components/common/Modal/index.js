import React, { Component, PropTypes } from 'react';
import Rodal from 'rodal';

// note: this component has a dependency on global styles
// import 'rodal/lib/rodal.css';

import { ModalContainer } from './styled-components';

class Modal extends Component {

  constructor(props, context) {
    super(props, context);
    this.closeModalOnEscapePress = this.closeModalOnEscapePress.bind(this);
  }

  componentWillUnmount(){
    // remove listener for closing modal on press of escape key
    document.removeEventListener("keydown", this.closeModalOnEscapePress);
  }

  closeModalOnEscapePress(event) {
    if (event.keyCode === 27) {
      this.props.onClose(event);
    }
  }

  render() {
    const { closeModalOnEscapePress } = this;
    const { children, visible, onClose, width, height, measure } = this.props;
    // stop the background from scrolling (if scroll bars are on the page)
    if (visible){
      // add listener for closing modal on press of escape key
      document.addEventListener("keydown", closeModalOnEscapePress);
      if ( window.innerWidth > document.documentElement.clientWidth ){
        document.body.classList.add('modal-is-open');
      }
    } else {
      // remove listener for closing modal on press of escape key
      document.removeEventListener("keydown", closeModalOnEscapePress);
      document.body.classList.remove('modal-is-open');
    }

    return (
        <Rodal visible={visible} onClose={onClose} width={width} height={height} measure={measure}>
          <ModalContainer>
            {children}
          </ModalContainer>
        </Rodal>
    );
  }
}

Modal.propTypes =  {
  visible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  measure: PropTypes.string,
  children: PropTypes.any.isRequired,
};

export default Modal;
