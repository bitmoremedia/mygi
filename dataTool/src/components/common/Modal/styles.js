import styled from 'styled-components';

export const ModalContainer = styled.div`
  overflow-y: scroll;
  overflow-x: hidden;
  height: 100%;
`;
