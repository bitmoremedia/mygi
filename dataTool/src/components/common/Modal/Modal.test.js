import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Modal from './index'

test('component renders without error', () => {
  const wrapper = mount(
    <Modal visible onClose={() => {}}>
      Modal Content
    </Modal>,
  )
  expect(wrapper.exists()).toBe(true)
})

// INTERACTIVITY TESTS

test('onClose function is fired when modal is closed', () => {
  const onClose = jest.fn()
  const wrapper = mount(
    <Modal visible onClose={onClose}>
      Modal Content
    </Modal>,
  )
  expect(onClose).not.toHaveBeenCalled()
  wrapper.find('.rodal-close').simulate('click')
  expect(onClose).toHaveBeenCalled()
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const component = renderer.create(
    <Modal visible onClose={() => {}}>
      Modal Content
    </Modal>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('modal hidden', () => {
  const component = renderer.create(
    <Modal visible={false} onClose={() => {}}>
      Modal Content
    </Modal>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
