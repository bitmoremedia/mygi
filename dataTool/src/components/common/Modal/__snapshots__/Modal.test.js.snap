exports[`test component renders as expected 1`] = `
<div
  className="rodal rodal-fade-enter "
  onAnimationEnd={[Function]}
  style={
    Object {
      "WebkitAnimationDuration": "300ms",
      "animationDuration": "300ms",
      "display": "",
    }
  }>
  <div
    className="rodal-mask"
    onClick={[Function]}
    style={Object {}} />
  <div
    className="rodal-dialog rodal-zoom-enter"
    style={
      Object {
        "WebkitAnimationDuration": "300ms",
        "animationDuration": "300ms",
        "height": "240px",
        "width": "400px",
      }
    }>
    <span
      className="rodal-close"
      onClick={[Function]} />
    <div
      className="CiQaJ">
      Modal Content
    </div>
  </div>
</div>
`;

exports[`test modal hidden 1`] = `
<div
  className="rodal rodal-fade-leave "
  onAnimationEnd={[Function]}
  style={
    Object {
      "WebkitAnimationDuration": "300ms",
      "animationDuration": "300ms",
      "display": "none",
    }
  }>
  <div
    className="rodal-mask"
    onClick={[Function]}
    style={Object {}} />
  <div
    className="rodal-dialog rodal-zoom-leave"
    style={
      Object {
        "WebkitAnimationDuration": "300ms",
        "animationDuration": "300ms",
        "height": "240px",
        "width": "400px",
      }
    }>
    <span
      className="rodal-close"
      onClick={[Function]} />
    <div
      className="CiQaJ">
      Modal Content
    </div>
  </div>
</div>
`;
